pkg/sink/kafka/client.go

```diff
@@ -21,4 +26,178 @@ type Client interface {
 	// Partitions returns the sorted list of
 	// all partition IDs for the given topic.
 	Partitions(topic string) ([]int32, error)
+	SyncProducer() (SyncProducer, error)
+	AsyncProducer() (AsyncProducer, error)
+	MetricRegistry() metrics.Registry
+	// Close closes the client
+	Close() error
+}
+
+// SyncProducer is the kafka sync producer
+type SyncProducer interface {
+	// SendMessage produces a given message, and returns only when it either has
+	// succeeded or failed to produce. It will return the partition and the offset
+	// of the produced message, or an error if the message failed to produce.
+	SendMessage(topic string, partitionNum int32,
+		key []byte, value []byte) error
+
+	// SendMessages produces a given set of messages, and returns only when all
+	// messages in the set have either succeeded or failed. Note that messages
+	// can succeed and fail individually; if some succeed and some fail,
+	// SendMessages will return an error.
+	SendMessages(topic string, partitionNum int32,
+		key []byte, value []byte) error
+
+	// Close shuts down the producer; you must call this function before a producer
+	// object passes out of scope, as it may otherwise leak memory.
+	// You must call this before calling Close on the underlying client.
+	Close() error
+}
+
+// AsyncProducer is the kafka async producer
+type AsyncProducer interface {
```

Comment:
is this interface only for sarama?

Reply:
for sarama and kafka-go, I will refine it in next PR